apache/superset
# feat: Adds support to multiple dependencies to the native filters

### SUMMARY
Adds support to multiple dependencies to the native filters. This will allow the user to create hierarchical and graph relationships among filters. The changes are the result of a series of interviews conducted by the design team with Superset users. Here's a summary of the changes:
- Creates the "Values are dependent on other filters" section
- Removes the "Filters is hierarchical" section
- Adds indentation to the collapsable containers
- Removes the hierarchical display of filters in the left panel of the modal
- Adjusts the algorithm to validate circular dependencies on the fly
- [Remove cascading popovers from filter bar](https://github.com/apache/superset/pull/18790) @kgabryje
- Effectively uses the [filter cards](https://github.com/apache/superset/pull/18874) to display filter context
- Adjusts RTL and Cypress the tests

It's important to note that major refactors are not in the scope at this stage. Our first objective is to deprecate FilterBox. We'll tackle code improvements in the next stage of the native filters revamp.

@kasiazjc @jess-dillard

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/155568831-ae7871b9-5c54-4060-beeb-11ac70978463.mov


### TESTING INSTRUCTIONS
We need to test the whole module. Apart from the basic tests, some important behaviors that should be checked:
- When selecting "Values are dependent on other filters" a value is automatically suggested and this value does not generate a circular dependency warning if possible.
- The circular dependency checking should occur on the fly while the user is selecting the filter dependencies.
- If a filter is removed or has its type changed to one different than "select", all references to this filter should display "(deleted or invalid filter)". After saving and reopening the modal, that reference should disappear.
- Only filters of type "select" should be available to be referenced as a dependency.
- All filters in scope should be displayed on the filter bar independently of their relationships.
- The values of a filter should be affected by its dependencies when applicable.
- The filter card context is correctly displayed and text is truncated as expected. Please refer to #18874 to see the requirements.

### ADDITIONAL INFORMATION
